contracts/lockdrop/src/contract.rs

```diff
@@ -2024,6 +2037,73 @@ pub fn query_lockup_info(
     })
 }
 
+/// @dev Returns pending asset rewards for a specified lockup position
+/// @param user_address : User address
+/// @param terraswap_lp_token : Pool identifier to identify the LP pool
+/// @param duration : Duration of the lockup
+pub fn query_pending_asset_reward(
+    deps: Deps,
+    env: Env,
+    user_address: String,
+    terraswap_lp_token: String,
+    duration: u64,
+) -> StdResult<PendingAssetRewardResponse> {
+    let user_address = addr_validate_to_lower(deps.api, &user_address)?;
+    let terraswap_lp_token = addr_validate_to_lower(deps.api, &terraswap_lp_token)?;
+
+    let lockup_key = (&terraswap_lp_token, &user_address, U64Key::new(duration));
+
+    let lockup_info_opt = LOCKUP_INFO
+        .may_load(deps.storage, lockup_key.clone())?
+        .filter(|lock_info| lock_info.astroport_lp_transferred.is_none());
+
+    let mut user_reward = Uint128::zero();
+    if let Some(lockup_info) = lockup_info_opt {
+        let pool_info = ASSET_POOLS.load(deps.storage, &terraswap_lp_token)?;
+        if !pool_info.has_asset_rewards {
+            return Err(StdError::generic_err("This pool does not have rewards"));
+        }
+
+        let MigrationInfo {
+            astroport_lp_token, ..
+        } = pool_info
+            .migration_info
+            .as_ref()
+            .ok_or_else(|| StdError::generic_err("The pool was not migrated to astroport"))?;
+        let pair_info = pair_info_by_pool(deps, astroport_lp_token.clone())?;
+        let pending_rewards: Asset = deps.querier.query_wasm_smart(
+            pair_info.contract_addr,
+            &astroport::pair_stable_bluna::QueryMsg::PendingReward {
+                user: env.contract.address.to_string(),
+            },
+        )?;
+
+        let reward_index = Decimal256::from_ratio(
+            Uint256::from(pending_rewards.amount),
+            pool_info.terraswap_amount_in_lockups,
+        );
+
+        let user_asset_reward_index = USERS_ASSET_REWARD_INDEX.key(lockup_key);
```

Comment:
Let's rename it user_asset_reward_path and it's ready!

Reply:
done